domain/personallearningplan/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/personallearningplan/Exceptions.kt

```diff
@@ -27,3 +22,28 @@ class ActionPlanAlreadyExistsException(message: String) : RuntimeException(messa
  */
 class GoalNotFoundException(val prisonNumber: String, val goalReference: UUID) :
   RuntimeException("Goal with reference [$goalReference] for prisoner [$prisonNumber] not found")
+
+/**
+ * Thrown when a Goal cannot be arhived/unarchived due to the current state.
+ */
+class InvalidGoalStateException(val prisonNumber: String, val goalReference: UUID, val status: String, val action: String) :
```

Comment:
Can we make this a little more type safe perhaps?
Maybe make the `status` field `uk.gov.justice.digital.hmpps.domain.personallearningplan.GoalStatus` ?
And the `action` field - could we make that one of the values in the companion object? (would have to be an enum I guess)